edb/server/bootstrap.py

```diff
@@ -928,6 +928,7 @@ async def _populate_data(schema, compiler, conn):
     script = '''
         INSERT stdgraphql::Query;
         INSERT stdgraphql::Mutation;
+        INSERT std::VirtualObject;
```

Comment:
It would be much better if we make each instance of `VirtualObject` distinct, i.e. have a different id.  Automatic injection of `id := uuid_generate_v1mc()` should do the trick.

Reply:
Hm, what's the rationale for caring about that? I don't expect anybody to really be looking at the id of the anonymous queries much, and so I don't see the advantage of adding more magic here, I guess, instead of having it be straightforward syntactic sugar?
Two other points:
 * We don't allow assigning to `id` as a computable now, but it should be too hard to override that for this case if we need to
 * Injecting a call to `uuid_generate_v1mc()` might cause it to need materialization

Reply:
> Hm, what's the rationale for caring about that?

Equality.

```
WITH
   Foo := {foo := 1}
   Bar := {bar := '2'}
SELECT Foo = Bar;
```

This should return `false`.

Reply:
Fair, though it seems a little marginal to me still.

It looks like just injecting the `id` wouldn't be sufficient, though; some pgsql-side work would be needed to actually believe that `id` is the object value

Reply:
> This should return false.

+1

Reply:
> some pgsql-side work would be needed to actually believe that `id` is the object value

Wouldn't materialization solve this naturally?

Reply:
> Wouldn't materialization solve this naturally?

That's a good point: I tried it out on my materialization branch, and it can make the computed `id` get viewed as the object value.

It then tried to join that subquery with the actual backing table (in order to ensure a source rvar), which resulted in an empty output. So if we want to generate fake ids dynamically, we'd need to ensure that we *never* try to join it against the real table for `std::VirtualObject`.

That's probably doable but I think it would be fragile and hacky.

Reply:
> That's probably doable but I think it would be fragile and hacky.

It doesn't have to be.  We can use the same machinery we use to do ACL-style filtering on type objects (grep `get_access_policy_filters`).

Reply:
Hm, on a second thought, that wouldn't really help here.

> That's probably doable but I think it would be fragile and hacky.

So, if we force materialization on all properties of the anon shape, there would technically be no need to join the actual "source" rvar there, right?  We just need to make sure we're not joining it frivolously, which we shouldn't be doing anyway for performance reasons.